src/microbe_stage/editor/MicrobeEditorGUI.cs

```diff
@@ -1346,6 +1297,89 @@ private void UpdateSymmetryIcon()
         }
     }
 
+    /// <summary>
+    ///   Lock / unlock a single organelle that need a nucleus
+    /// </summary>
+    private void UpdatePartAvailability(bool hasNucleus, OrganelleDefinition organelle)
+    {
+        // Item could be a non-implemented organelle so check if it really exists first.
+        if (!placeablePartSelectionElements.ContainsKey(organelle))
+        {
+            // Given organelle has no part selection associated.
+            return;
+        }
+
+        var item = placeablePartSelectionElements[organelle];
+
+        if (item.Name == nucleus.InternalName)
+        {
+            item.Locked = hasNucleus;
+        }
+        else if (organelle.RequiresNucleus)
+        {
+            item.Locked = !hasNucleus;
+        }
+        else
+        {
+            item.Locked = false;
+        }
+    }
+
+    /// <summary>
+    ///   Associates all existing cell part selections with their respective part types based on their Node names.
+    /// </summary>
+    private void SetupMicrobePartSelections()
+    {
+        var organelleSelections = GetTree().GetNodesInGroup(
+            "PlaceablePartSelectionElement").Cast<MicrobePartSelection>().ToList();
+        var membraneSelections = GetTree().GetNodesInGroup(
+            "MembraneSelectionElement").Cast<MicrobePartSelection>().ToList();
+
+        foreach (var entry in organelleSelections)
+        {
+            // Special case with registering the tooltip here for item with no associated organelle
+            entry.RegisterToolTipForControl(ToolTipManager.Instance.GetToolTip(
+                entry.Name, "organelleSelection"), tooltipCallbacks);
+
+            if (!SimulationParameters.Instance.DoesOrganelleExist(entry.Name))
+                continue;
+
+            var organelle = SimulationParameters.Instance.GetOrganelleType(entry.Name);
+
+            // Only add items with valid organelles to dictionary
+            placeablePartSelectionElements.Add(organelle, entry);
+
+            if (!entry.IsConnected(
+                nameof(MicrobePartSelection.OnPartSelected), this, nameof(OnOrganelleToPlaceSelected)))
+            {
+                entry.Connect(
+                    nameof(MicrobePartSelection.OnPartSelected), this, nameof(OnOrganelleToPlaceSelected));
+            }
+        }
+
+        foreach (var entry in membraneSelections)
+        {
+            // Special case with registering the tooltip here for item with no associated membrane
+            entry.RegisterToolTipForControl(ToolTipManager.Instance.GetToolTip(
+                entry.Name, "membraneSelection"), tooltipCallbacks);
+
+            if (!SimulationParameters.Instance.DoesMembraneExist(entry.Name))
+                continue;
```

Comment:
I think it would make sense to ensure that a part is locked here before the continue just incase some unimplemented part is not locked by default.